### ISSUE TITLE
typesafe onMutation promise

### ISSUE THREAD

**Issue by skk2142** [relay-tools/relay-hooks#132]
`onMutation` hook resolves with the mutation response or reject with any error (except when an onError callback is specified, in which case it will resolve with no value on errors). 

Wouldn't it be a good idea to make the resolve value optional regardless of whether `onError` is specified or nor? Currently, if I specify `onError`, I still get the typescript type of resolve value as non-null and this could lead to runtime error.

**Comment by morrys** [comment]
Hi @skk2142, I would like to add this issue to the new release (4.0.0).

The solution that could be adopted is to change this:
https://github.com/relay-tools/react-relay-mutation/blob/master/src/index.tsx#L90-L95
```ts
if (mergedConfig.onError) {
  mergedConfig.onError(error);
  resolve();
} else {
  reject(error);
}
```
in
```ts
if (mergedConfig.onError) {
  mergedConfig.onError(error);
} 
reject(error);
```
and remove support for HOC Mutation.

**Comment by morrys** [comment]
Issue resolved with relay-hooks version [4.0.0](https://github.com/relay-tools/relay-hooks/releases/tag/v4.0.0)
